pkg/controllers/clusterresourceplacement/controller.go

```diff
@@ -133,35 +137,68 @@ func (r *Reconciler) deleteClusterResourceSnapshots(ctx context.Context, crp *fl
 // If the error type is ErrUnexpectedBehavior, the controller will skip the reconciling.
 func (r *Reconciler) handleUpdate(ctx context.Context, crp *fleetv1beta1.ClusterResourcePlacement) (ctrl.Result, error) {
 	revisionLimit := fleetv1beta1.RevisionHistoryLimitDefaultValue
+	crpKObj := klog.KObj(crp)
+	oldCRP := crp.DeepCopy()
 	if crp.Spec.RevisionHistoryLimit != nil {
 		revisionLimit = *crp.Spec.RevisionHistoryLimit
 		if revisionLimit <= 0 {
 			err := fmt.Errorf("invalid clusterResourcePlacement %s: invalid revisionHistoryLimit %d", crp.Name, revisionLimit)
-			klog.ErrorS(controller.NewUnexpectedBehaviorError(err), "Invalid revisionHistoryLimit value and using default value instead", "clusterResourcePlacement", klog.KObj(crp))
+			klog.ErrorS(controller.NewUnexpectedBehaviorError(err), "Invalid revisionHistoryLimit value and using default value instead", "clusterResourcePlacement", crpKObj)
 			// use the default value instead
 			revisionLimit = fleetv1beta1.RevisionHistoryLimitDefaultValue
 		}
 	}
 
-	_, err := r.getOrCreateClusterSchedulingPolicySnapshot(ctx, crp, int(revisionLimit))
+	latestSchedulingPolicySnapshot, err := r.getOrCreateClusterSchedulingPolicySnapshot(ctx, crp, int(revisionLimit))
 	if err != nil {
 		return ctrl.Result{}, err
 	}
-	selectedResources, _, err := r.selectResourcesForPlacement(crp)
+	selectedResources, selectedResourceIDs, err := r.selectResourcesForPlacement(crp)
 	if err != nil {
 		return ctrl.Result{}, err
 	}
 	resourceSnapshotSpec := fleetv1beta1.ResourceSnapshotSpec{
 		SelectedResources: selectedResources,
 	}
-	_, err = r.getOrCreateClusterResourceSnapshot(ctx, crp, &resourceSnapshotSpec, int(revisionLimit))
+	latestResourceSnapshot, err := r.getOrCreateClusterResourceSnapshot(ctx, crp, &resourceSnapshotSpec, int(revisionLimit))
 	if err != nil {
 		return ctrl.Result{}, err
 	}
+	if err := r.setPlacementStatus(ctx, crp, selectedResourceIDs, latestSchedulingPolicySnapshot, latestResourceSnapshot); err != nil {
+		return ctrl.Result{}, err
+	}
 
-	// update the status based on the latestPolicySnapshot status
-	// update the status based on the work
-	return ctrl.Result{}, nil
+	if err := r.Client.Status().Update(ctx, crp); err != nil {
+		klog.ErrorS(err, "Failed to update the status", "clusterResourcePlacement", crpKObj)
+		return ctrl.Result{}, err
+	}
+	klog.V(2).InfoS("Updated the clusterResourcePlacement status", "clusterResourcePlacement", crpKObj)
+
+	if isRolloutCompleted(crp) {
```

Comment:
I think we probably also need some tests on the event get triggered. Can leave this as UT task for later.